### PR TITLE
Loaded GeoJSON file and renders map of it.

### PR DESCRIPTION
**Pull request by NotakingButaGod**
Map rendering of GeoJSON file, includes handling in the fileloadview and mapdisplay

### CODE REVIEW

File: src/Views/FileLoaderView.tsx [hunk 1]
```diff
@@ -1,15 +1,11 @@
...
+import GeoJSONMapDisplay from './GeoJSONMapDisplay';
 import '../css/FileLoader.css';
 
-export default function FileLoaderView({ children, setGeoPackage }: { children: any, setGeoPackage: (geoPackage: GeoPackage) => void }) {
+export default function FileLoaderView({ children, setGeoPackage }: { children: any, setGeoPackage: (geoPackage: File) => void }) {
 
     const fileInputRef = useRef<HTMLInputElement>(null);
-
-
+    const [geoJsonData, setGeoJSON] = useState<any>(null);
```
Comment (MingkaiChen) [inline src/Views/FileLoaderView.tsx:8]: Please do not use *geoJSON* terms here. Also, do not use *any* if possible.

File: src/Views/FileLoaderView.tsx [hunk 2]
```diff
@@ -19,8 +15,20 @@ export default function FileLoaderView({ children, setGeoPackage }: { children:
...
+            reader.onload = async (e) => {
+                const content = e.target?.result;
+                try {
+                    const parsedGeoJson = JSON.parse(content as string);
+                    setGeoJSON(parsedGeoJson)
+                }catch (error) {
+                    console.error("Error parsing GeoJSON:", error);
+                }
+            };
+            reader.readAsText(file);
```
Comment (MingkaiChen) [inline src/Views/FileLoaderView.tsx:31]: Should be conditional logic here to determine all the types we need to support and read them with the appropriate methods.

**Review by MingkaiChen — COMMENTED** [review]
Please review comments carefully.

**Review by MingkaiChen — COMMENTED** [review]
Generally looks fine. Keep working on shapefiles.

**Review by MingkaiChen — APPROVED** [review]
Looks good. Approved PR.

File: src/Views/MapWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                        if (!groupedFiles[basename]) {
+                            groupedFiles[basename] = {};
+                        }
+                        const ext = (relativePath.split('.').pop() ?? "").toLowerCase();
+                        groupedFiles[basename][ext] = zipEntry;
+                    });
+                    // Extract and parse all the shp and dbf files
+                    let geojsonDatas : GeoJsonObject[] = [];
+                    for (let basename in groupedFiles) {
+                        if (groupedFiles[basename].shp && groupedFiles[basename].dbf) {
```
Comment (0xuana) [inline src/Views/MapWrapper.tsx:58]: .zip upload, if dbf does not exist, covert return error.
If we want to convert a single shp file successfully, we need to add another shp condition.

File: src/Views/MapWrapper.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                        console.error("Error parsing GeoJSON:", error);
+                    }
+                }
+                // handle kml file 
+                else if (fileType === "kml") {
+                    const kmlDocument = new domParser.DOMParser().parseFromString(content as string, 'text/xml');
+                    const convertedGeoJSON = tj.kml(kmlDocument);
+                    const geoJsonArray: GeoJsonObject[] = [convertedGeoJSON];
+                    setGeoJSONData(geoJsonArray);
+                }
```
Comment (0xuana) [inline src/Views/MapWrapper.tsx:40]: *handle* shapefile zip will be more accurate.

File: src/Views/MapWrapper.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                        const basename = relativePath.split('.').slice(0, -1).join('.');
+                        if (!groupedFiles[basename]) {
+                            groupedFiles[basename] = {};
+                        }
+                        const ext = (relativePath.split('.').pop() ?? "").toLowerCase();
+                        groupedFiles[basename][ext] = zipEntry;
+                    });
+                    // Extract and parse all the shp and dbf files
+                    let geojsonDatas : GeoJsonObject[] = [];
+                    for (let basename in groupedFiles) {
```
Comment (0xuana) [inline src/Views/MapWrapper.tsx:57]: .zip upload, if dbf does not exist, covert return error.
If we want to convert a single shp file successfully, we need to add another shp condition. (same layer as kml, geojson, zip file test. 
Moreover, this makes a single shapefile in zip that cannot convert successfully. Maybe we can add an else if condition to handle **only shapefile** occur in zipcondition.

Comment (0xuana) [reply src/Views/MapWrapper.tsx:57]: single shp cannot convert into geojson, due to the location information loss (large countries may can correct by package)

**Review by 0xuana — APPROVED** [review]
File header check and pass check result to *MapWrapper* to convert it into *geojson*
*shapefile* package can be deleted in *package.json*

**Review by MingkaiChen — APPROVED** [review]
Final Approve.

**Comment by MingkaiChen** [comment]
Resolved #9, #10 and #11.

**Review by 0xuana — COMMENTED** [review]
add shp condition to zip/shp condition

**Pull request merged**